libs/atqr/domain/src/lib/challenge/challenge.entity.ts

```diff
@@ -29,30 +29,32 @@ export class Challenge {
   private _status?: ChallengeStatus;
 
   constructor(
-    private _goal: string,
-    private _supervisorName: string,
-    private _supervisorEmail: EmailAddress,
-    private _player: Player,
-    price: number,
-    deadline: Date,
-    private _paymentMethod?: PaymentMethodEntity,
-    _status: ChallengeStatus = ChallengeStatus.Ongoing,
-    private _supervisorStatus: SupervisorEnum = SupervisorEnum.notInvited
+    private _goalId: GoalId,
+    private _owner: User,
+    private _invitees: Invitee[],
+    private _deadline: Date,
+    private _pledge: number,
+    private _requiredProof?: string,
+    private _enrollmentDeadline: Date
   ) {
-    this._id = new Guid();
-    if (price >= 25) {
-      this._price = price;
+    // challengeId is in challenge's updated proprieties on issue 120, already pull requested
+    this._challengeId = new ChallengeId();
```

Comment:
in the constructor it is a goal Id, not challenge Id.

Reply:
@rairezende

Reply:
I'm just comparing it with the challenge constructor in the domain-spec > aggregates file, since that's what your issue asks for and that is how it is in there. That being said I agree on the challenge needing it's own id, so you may add it there, but the property "goal id" also exists and its id is "goalId".

Reply:
@albnunes